external-crates/move/move-bytecode-verifier/src/stack_usage_verifier.rs

```diff
@@ -59,6 +59,11 @@ impl<'a> StackUsageVerifier<'a> {
             let (num_pops, num_pushes) = self.instruction_effect(&code[i as usize])?;
             if let Some(new_pushes) = u64::checked_add(overall_push, num_pushes) {
                 overall_push = new_pushes
+            } else {
+                if config.max_push_size.is_some() {
```

Comment:
why is the error gated on `max_push_size`?

Reply:
We only care about limiting if a limit is specified in the config

Reply:
I do not think so, this is an overflow right? not a limit problem.
Thoughts?

Reply:
Coming from the main branch PR: https://github.com/move-language/move/pull/1028

I feel like @oxade is correct that we should only return VALUE_STACK_PUSH_OVERFLOW if the limit is set in the config. However, in case the limit is not set, we still need to deal with the potential overflow, even though I doubt if this can actually happen in practice due to the use of u64.